src/actions/vstudio/vs2010_vcxproj.lua

```diff
@@ -1231,7 +1231,15 @@
 
 
 	function m.generateDebugInformation(cfg)
-		m.element("GenerateDebugInformation", nil, tostring(cfg.flags.Symbols ~= nil))
+		if cfg.flags.Symbols and _ACTION >= "vs2015" then
+			if cfg.flags.FastLink then
+				m.element('GenerateDebugInformation', nil, "DebugFastLink")
+			else
+				m.element('GenerateDebugInformation', nil, "Debug")
```

Comment:
@tvandijck I believe the solution is for all the actions to translate the values to their respective set of "allowed" values. For example, `optimize` supports six values and [gcc.lua](https://github.com/premake/premake-core/blob/b4bc07f4ed6ccdb8f94b4695283fda11203df4b7/src/tools/gcc.lua#L61-L68), [snc.lua](https://github.com/premake/premake-core/blob/b883c7f45f7e08b39effcf7d174833a67ee1133e/src/tools/snc.lua#L21-L28), [dotnet.lua](https://github.com/premake/premake-core/blob/b883c7f45f7e08b39effcf7d174833a67ee1133e/src/tools/dotnet.lua#L210-L214) and [msc.lua](https://github.com/premake/premake-core/blob/b883c7f45f7e08b39effcf7d174833a67ee1133e/src/tools/msc.lua#L56-L63) all translate the values differently and not just to toolset specific values, `Off` and `Debug` are the same in msc, 'Full' and 'Speed' are the same for gcc, only three are supported in dotnet but they're all the same and they're all different for snc.

@TurkeyMan was telling me to do this for the premake-android module PR I opened and it makes sense as it allows premake user scripts to be OS, toolset and action agnostic as much as possible. All we, as users, need to do is give premake a single value and those poor premake developers need to figure out what that means for each generator. :stuck_out_tongue: